src/transactionNext.spec.ts

```diff
@@ -0,0 +1,223 @@
+import { assert } from "chai";
+import { TestWallet, loadTestWallets } from "./testutils";
+import { TransactionNext, TransactionComputer, Transaction } from "./transaction";
+import BigNumber from "bignumber.js";
+import { ProtoSerializer } from "./proto";
+
+class NetworkConfig {
+    MinGasLimit: number;
+    GasPerDataByte: number;
+    GasPriceModifier: number;
+    ChainID: string;
+
+    constructor(minGasLimit: number = 50000) {
+        this.MinGasLimit = minGasLimit;
+        this.GasPerDataByte = 1500;
+        this.GasPriceModifier = 0.01;
+        this.ChainID = "D";
+    }
+}
+
+describe("test transaction next", async () => {
+    let wallets: Record<string, TestWallet>;
+    const networkConfig = new NetworkConfig();
+    const transactionComputer = new TransactionComputer();
+
+    before(async function () {
+        wallets = await loadTestWallets();
+    });
+
+    it("should serialize transaction for signing", async () => {
+        const sender = "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th";
+        const receiver = "erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx";
+
+        let transaction = new TransactionNext({
+            chainID: networkConfig.ChainID,
+            sender: sender,
+            receiver: receiver,
+            gasLimit: 50000,
+            value: 0,
+            version: 2,
+            nonce: 89,
+        });
+
+        let serializedTransactionBytes = transactionComputer.computeBytesForSigning(transaction);
+        let serializedTransaction = Buffer.from(serializedTransactionBytes).toString();
+
+        assert.equal(
+            serializedTransaction,
+            `{"nonce":89,"value":"0","receiver":"erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx","sender":"erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th","gasPrice":1000000000,"gasLimit":50000,"chainID":"D","version":2}`
```

Comment:
I would extract the transaction payload

```
const transactionPayload = {
      chainID: networkConfig.ChainID,
      sender: sender,
      receiver: receiver,
      gasLimit: 50000,
      value: 0,
      version: 2,
      nonce: 89,
  }
  let transaction = new TransactionNext(transactionPayload);
// ...
        
```
 and in this asset I'd use: 
 
```
 assert.equal(
  serializedTransaction,
  JSON.stringify(transactionPayload)
);
```

Reply:
Well, here I don't think it's that good to do that because the serialized transaction has a specific order and in order to match that with `JSON.stringify()` you'd have to specify the elements in the plain object the same as in the serialized transaction.